configure.ac

```diff
@@ -855,6 +855,32 @@ if test "${with_crypto_library}" = "openssl"; then
 		AC_DEFINE([HAVE_OPENSSL_ENGINE], [1], [OpenSSL engine support available])
 	fi
 
+	# Check whether OpenSSL provider support is usable -- need version >= 3.0.1
+	have_usable_openssl_provider="yes"
+	AC_MSG_CHECKING([Check if OpenSSL version >= 3.0.1])
+	AC_COMPILE_IFELSE(
+		[AC_LANG_PROGRAM(
+			[[
+#include <openssl/opensslv.h>
+			]],
+			[[
+/*     Version encoding: MNN00PPS - see opensslv.h for details */
+#if OPENSSL_VERSION_NUMBER < 0x30000010L
+#error OpenSSL with no or buggy provider support
+#endif
+			]]
+		)],
+		[AC_MSG_RESULT([ok])],
+		[
+			have_usable_openssl_provider="no"
+			AC_MSG_RESULT([no])
+		]
+	)
+
+	if test "${have_usable_openssl_provider}" = "yes"; then
+		AC_DEFINE([HAVE_XKEY_PROVIDER], [1], [Enable external key loading provider])
+	fi
+
```

Comment:
Yes, we could  use `#ifdef OPENSSL_VERSION_NUMBER >= 30000010L` and not have this define. Using a separate variable looked cleaner to me instead of having to write the version number in every if def. Also easier to change if it has to be fine-tuned later or if someone wants a build without xkey_provider, easier to change configure than the source.
But I'm okay with either approach.

Reply:
I would then prfer to have the 
```
#ifdef OPENSSL_VERSION_NUMBER >= 30000010L
#define HAVE_XKEY_PROVIDER
#endif
```

approach in a header. That still allows kind of easy compiling without xkey if needed while not adding extra logic to autoconf.

Reply:
okay, let me see whether this could be tucked-in xkey_common.h

Reply:
I also have an AM_CONDITIONAL with the same name for use in Makefile.am for tests. Can't think of a way to do conditional compiling of the test without it --- can make the test empty when HAVE_XKEY_PROVIDER is undefined but that's not ideal.